outlook/mail/main.go

```diff
@@ -121,26 +136,62 @@ func main() {
 	}
 }
 
-
 func smartSplit(s, sep string) []string {
 	if s == "" {
 		return []string{} // Return an empty slice if the input is empty
 	}
 	return strings.Split(s, sep)
 }
 
-
-func getDraftInfoFromEnv() graph.DraftInfo {
+func getDraftInfoFromEnv(ctx context.Context) (graph.DraftInfo, error) {
 	var attachments []string
 	if os.Getenv("ATTACHMENTS") != "" {
 		attachments = smartSplit(os.Getenv("ATTACHMENTS"), ",")
 	}
 
+	body := os.Getenv("BODY")
+	replyMessageID := os.Getenv("REPLY_MESSAGE_ID")
+	replyAll := os.Getenv("REPLY_ALL") == "true"
+	cc := os.Getenv("CC")
+
+	if replyMessageID != "" {
+		trueMessageID, err := id.GetOutlookID(ctx, replyMessageID)
+		if err != nil {
+			return graph.DraftInfo{}, fmt.Errorf("failed to get true message ID: %w", err)
+		}
+
+		c, err := client.NewClient(global.ReadOnlyScopes)
+		if err != nil {
+			return graph.DraftInfo{}, fmt.Errorf("failed to create client: %w", err)
+		}
+
+		result, err := graph.GetMessageDetails(ctx, c, trueMessageID)
+		if err != nil {
+			return graph.DraftInfo{}, fmt.Errorf("failed to get message details: %w", err)
+		}
+
+		// Format reply in Outlook web client style
+		originalMessage := fmt.Sprintf("<br><br>**-------- Original Message --------**  <br><br>**From:** %s  <br>**Sent:** %s  <br>**To:** %s  <br>**Subject:** %s  <br><br>%s  <br><br>",
+			*result.GetFrom().GetEmailAddress().GetAddress(),
+			result.GetReceivedDateTime().Format("Monday, January 2, 2006 3:04 PM"),
+			*result.GetToRecipients()[0].GetEmailAddress().GetAddress(),
+			*result.GetSubject(),
+			*result.GetBodyPreview())
```

Comment:
Should this contain the full body? or is the preview enough?

Reply:
I think preview is probably fine now... can tweak that later